From d4a8be56499162d76858a3ab65053435174c48d1 Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.krummenacher@toradex.com>
Date: Wed, 17 Jun 2020 09:14:20 +0000
Subject: [PATCH] g2d-renderer: ignore G2D_TILED_STATUS

weston 8.0.0.imx uses a G2D_TILED_STATUS feature which is unavailable
in older NXP BSPs. (4.14.98_2.3.0).

Remove the relevant code from compilation.

Signed-off-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 libweston/renderer-g2d/g2d-renderer.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libweston/renderer-g2d/g2d-renderer.c b/libweston/renderer-g2d/g2d-renderer.c
index 34a077b7..b9ede558 100644
--- a/libweston/renderer-g2d/g2d-renderer.c
+++ b/libweston/renderer-g2d/g2d-renderer.c
@@ -477,6 +477,10 @@ get_g2dSurface(struct wl_viv_buffer *buffer, struct g2d_surfaceEx *g2dSurface)
 	g2dSurface->base.height = buffer->height;
 	g2dSurface->base.rot	= G2D_ROTATION_0;
 
+#if 1
+	/* ignore tileStatus on i.MX8MM and i.MX6 */
+	buffer->tileStatus_enabled = 0;
+#else
 	if(buffer->ts_addr && buffer->tileStatus_enabled)
 	{
 		g2dSurface->tiling             |= G2D_TILED_STATUS;
@@ -485,6 +489,7 @@ get_g2dSurface(struct wl_viv_buffer *buffer, struct g2d_surfaceEx *g2dSurface)
 		g2dSurface->ts.fc_value        = buffer->fcValue;
 		g2dSurface->ts.fc_value_upper  = buffer->fcValueUpper;
 	}
+#endif
 }
 
 static void
-- 
2.20.1

